app/services/pairing_strategies/plain_pairing.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# PlainPairing is a Plain Ruby model object that provides a Pairing representation
+# without the full overhead of the ActiveRecord Pairing object. This exists to keep
+# the pairing logic light on memory and object allocation / GC overhead.
+module PairingStrategies
+  class PlainPairing
+    def initialize(player1, player1_score, player2, player2_score)
+      @player1 = player1
+      @player1_score = player1_score
+      @player2 = player2
+      @player2_score = player2_score
+    end
+
+    attr_accessor :player1, :player1_score, :player2, :player2_score, :player1_side, :table_number
+
+    def players
+      [player1, player2]
+    end
+
+    def bye?
+      player1.nil? or player2.nil?
+    end
+
+    def fixed_table_number?
+      players.compact.any?(&:fixed_table_number?)
+    end
+
+    def fixed_table_number
+      fixed_table_number? ? players.compact.filter(&:fixed_table_number?).map(&:fixed_table_number).min : nil
```

Comment:
Calling min on an empty enumerable returns nil, so I don't think there's any need to call `fixed_table_number?` at the beginning here?

Reply:
thanks, that tidies this up a bit.